crates/wasi-provider/src/states/running.rs

```diff
@@ -0,0 +1,53 @@
+use k8s_openapi::api::core::v1::Pod as KubePod;
+use kube::api::{Api, PatchParams};
+use kubelet::container::Status;
+use kubelet::state::prelude::*;
+use log::error;
+
+use super::error::Error;
+use super::terminated::Terminated;
+use crate::PodState;
+
+state!(
+    /// The Kubelet is running the Pod.
+    Running,
+    PodState,
+    {
+        let client: Api<KubePod> = Api::namespaced(
+            kube::Client::new(pod_state.shared.kubeconfig.clone()),
+            pod.namespace(),
+        );
+        let mut completed = 0;
+        let total_pods = pod.containers().len();
+
+        while let Some((name, status)) = pod_state.run_context.status_recv.recv().await {
+            if let Err(e) = client
```

Comment:
One of my goals of the state machine was to eliminate any pod status patches outside of the context of the state machine. Can this be handled by transitioning to an error state or adding an error message to the Terminated state?

Reply:
If my hazy memory serves right... This was necessary to send container status updates to the API. `make_status` does not have a way to send patch updates, which is necessary for the e2e tests to pass.

Reply:
https://github.com/bacongobbler/krustlet/blob/02d15a8e6104b885a6704c4f8b4eec0722cbe36f/crates/kubelet/src/pod/status.rs#L18-L19

Reply:
But yes, the eventual goal is to eliminate these pod status patches from the control loop entirely.

Reply:
I do agree we need a better story around container status updates.

Reply:
I think the current idea is to make each container have its own state machine (separate from the pod one) and those state machines would take care of updating the status